### PR TITLE
[js-cookie] Fix signature of CookieReadConverter, improve types+tests

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/js-cookie/js-cookie
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Fixes #59418

### CODE REVIEW

File: types/js-cookie/index.d.ts [hunk 1]
```diff
@@ -86,16 +86,16 @@ declare namespace Cookies {
...
+    interface Converter<TConv> {
         write?: CookieWriteConverter<TConv> | undefined;
-        read?: CookieReadConverter | undefined;
+        read?: CookieReadConverter<TConv> | undefined;
     }
 
-    type CookieWriteConverter<T extends object> = (value: string | T, name: string) => string;
-    type CookieReadConverter = (value: string, name: string) => string;
+    type CookieWriteConverter<T> = (value: string | T, name: string) => string;
+    type CookieReadConverter<T> = (value: string, name: string) => T | string;
```
Comment (peterblazejewicz) [inline types/js-cookie/index.d.ts:98]:
```suggestion
    type CookieReadConverter<T> = (value: string, name: string) => string | T;
```
just for consistency with 'string | T' used everywhere else.

**Comment by peterblazejewicz** [comment]
is this for a use case like:
```ts
// $ExpectType CookiesStatic<Person>
Cookies.withConverter<Person>({
    // $ExpectType (value: string | Person, name: string) => string
    write: (value, name) => {
        return JSON.stringify(value);
    },
    // $ExpectType (value: string, name: string) => Person
    read: (value, name) => {
        return JSON.parse(value) as Person;
    },
});
```
that is, there is no a test case for the use case.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@BendingBender thx!

**Comment by BendingBender** [comment]
Ready to merge

**Pull request merged**